oxidation/crates/parsec_api_protocol/src/transport.rs

```diff
@@ -0,0 +1,69 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use async_std::net::TcpStream;
+use async_tungstenite::{
+    accept_async, client_async, tungstenite,
+    tungstenite::{handshake::client::generate_key, Message},
+    WebSocketStream,
+};
+use futures::{
+    stream::{SplitSink, SplitStream},
+    SinkExt, StreamExt,
+};
+use http::{Request, Uri};
+
+pub struct Transport {
+    pub write: SplitSink<WebSocketStream<TcpStream>, Message>,
+    pub read: SplitStream<WebSocketStream<TcpStream>>,
+}
+
+// Only because we need user-agent
+fn build_request_from_uri<T: AsRef<str>>(uri: T) -> Request<()> {
+    dotenv::dotenv().ok();
+    let version = std::env::var("VERSION").unwrap();
```

Comment:
To get the version, in theory we should be using `CARGO_PKG_VERSION` env var that is set automatically by cargo during compilation.
However this won't work given we want the version of `libparsec` and not of `parsec_api_protocol` crate.
I think the easiest way to fix this is to have a dead simple `parsec_version` crate (to avoid cyclic dependencies) that will contain a const version field.

Reply:
So we can have the `parsec_version` crate going yolo and parse the `parsec/_version.py` file.
This is not as crazy as it seems given 1) the data we want to extract is very simple so a `^__version__ = "(.+)"$` regex is enough and 2) a failing the compilation if the regex search doesn't succeed should be good enough detection if the `_version.py` is modified in an unexpected way

note: we can improve the regex to `^__version__ = "(v[0-9]+\.[0-9]+\.[0-9]+.*)"$` if we want to be more pedantic, but I'm not sure it is worth it...